app/harvesters/loc_harvester.rb

```diff
@@ -0,0 +1,174 @@
+    opts[:uri] ||= ['https://www.loc.gov/collections/american-revolutionary-war-maps/?fo=json',
+      'https://www.loc.gov/collections/civil-war-maps/?fo=json',
+      'https://www.loc.gov/collections/panoramic-maps/?fo=json']
+    # TODO add name (check)
+    opts[:name] ||= 'loc'
+    # Item URI
+    opts[:item_uri] ||= 'https://www.loc.gov/item/'
+    super
+  end
+
+  ##
+  # @return [Hash] A hash documenting the allowable options to pass to
+  #   initializers.
+  #
+  # @see Krikri::Harvester::expected_opts
+  def self.expected_opts
+    {
+      key: :api,
+        opts: {
+       }
+    }
+  end
+
+  ##
+  # @param identifier [#to_s] the identifier of the record to get
+  # @return [#to_s] the record
+  def get_record(identifier)
+    response = request(opts[:item_uri] + identifier.to_s + '?fo=json')
+    build_record( response )
+  end
+
+
+  private
+
+  ##
+  # @param doc [Hash] the partial item content included in the search result.
+  # This content does not include the item's id, only URIs to the various
+  # item pages. Aggregate all the URIs and select those that match the
+  # JSON-enabled view (www.loc.gov/item/<id>)
+  #
+  # @return [Hash] the complete record
+  def get_item(doc)
+    uris = Array.new( [ doc['aka'], doc['id'], doc['url'] ] ).flatten
+    .uniq.grep(/www.loc.gov\/item/)
+    request(uris.first + '?fo=json') if !uris.empty?
+  end
+
+  ##
+  # Overrides ApiHarvester.get_identifier because the idenitifer field
+  # for Library of Congress
+  #
+  # @param doc [Hash] a JSON serialziation with an identifier
+  #
+  # @return [String] the provider's identifier for the document
+  def get_identifier(doc)
+    # doc is the response to a request for /item/<id>
+    doc['item']['library_of_congress_control_number']
+  end
+
+  ##
+  # @param response [Hash] a response from the LoC API
+  #
+  # @return [Integer] value of the number of records in the response
+    def get_count(response)
+      response['pagination']['of']
+    end
+  ##
+  # @param response [Hash] a response from the LoC API
+  #
+  # @return [Array] the documents in the response
+  def get_docs(response)
+    response['results']
+  end
+
+  ##
+  # Get the URL for the next request.
+  #
+  # @param opts [Hash] the response to the current request. The next page is
+  # provided in the ['next'] element
+  #
+  # @param response [Hash]
+  #
+  # @return [Hash] the next request's options hash
+  def next_options(response)
+    response['pagination']['next']
+  end
+
+  ##
+  # @return [Enumerator] an enumerator over the records
+  def enumerate_records
+    Enumerator.new do |yielder|
+      # For each of the sets
+      uri.each do | request_opts |
+        loop do
+          break if request_opts.nil?
+          response = request(request_opts)
+          docs = get_docs(response)
+          break if docs.empty?
+
+          # Queue #get_items threads
+          threads = []
+          docs.each do |r|
```

Comment:
I'm worried that this model (one thread per doc) is going to have a lot of overhead in managing the threads. 

With WebMock on, but no mocks, calls to `#records` and `#record_ids` simply hang somewhere in this code; this looks like it could be the culprit. Either way, I think we should probably be capping the thread count in the 10s, partitioning `docs` and passing off a portion to each thread.